### ISSUE TITLE
The maximum label length does not take effect

### ISSUE THREAD

**Issue by Akery** [kisvegabor/lvgl#3420]
### Perform all steps below and tick them with [x]
- [ ] Read the [FAQ](https://docs.lvgl.io/master/intro/index.html#faq)
- [ ] Check the related part of the [Documentation](https://docs.lvgl.io/)
- [ ] Update lvgl to the latest version
- [ ] Reproduce the issue in a [Simulator](https://docs.lvgl.io/latest/en/html/get-started/pc-simulator.html)

### Describe the bug

This method named 'lv_obj_set_style_max_height'  does not take effect when we set 'LV_LABEL_LONG_DOT'.

### To Reproduce

```
lv_obj_t* test = lv_label_create(lv_scr_act());
lv_label_set_text(test,"asfsdfasdfasdfasdfasdfashcvhhghgfhgfhgfhgfhfghfhfjhfhjfjhfghfjhfjhgtyedrtdcvkjhvjhvkjhvgydcydjghjvbjkfyufukyfhbkjhlhuilfgvhykfuyukdfasdfasfasfasfsdgasdfhjlkhajkfhlasjkhflksjhfsafsadfasdfadsfasdfasdfasdfasdfadsfasdfaasdfasdfdasdfasdfffasdfa");
lv_obj_set_width(test, 200);
lv_label_set_long_mode(test, LV_LABEL_LONG_DOT);
lv_obj_set_style_max_height(test, 340, LV_PART_MAIN);
```
### Screenshots or video

![image](https://user-images.githubusercontent.com/12009446/174761608-8a7fa9cc-a19a-46f8-9db7-1af994a9f8b1.png)

**Comment by kisvegabor** [comment]
I've tested it and it really doesn't work correctly. 
I think the expected behavior would be to show the text without dots until reaching the max height and start to show dots if the text height is about to grow beyond that max height. Do you agree? 

BTW, I don't think it was planned to make max-height work in this case, but I'd really nice to make it work.

**Comment by Akery** [comment]
I agree with you, and that's what I'm trying to achieve now.

**Comment by kisvegabor** [comment]
Sorry for the late reply. Unfortunately I can't see a straight forward way of fixing it now,  The sizing, overflow handling mechanisms of `lv_label` will be reworked in v9. As part of it this issue should be solved too.

**Comment by xaowang96** [comment]
@Akery I added a function call that solve this problem, like this

```
    lv_obj_t* test = lv_label_create(lv_scr_act());
    lv_label_set_text(
        test,
        "asfsdfasdfasdfasdfasdfashcvhhghgfhgfhgfhgfhfghfhfjhfhjfjhfghfjhfjhgtye"
        "drtdcvkjhvjhvkjhvgydcydjghjvbjkfyufukyfhbkjhlhuilfgvhykfuyukdfasdfasfa"
        "sfasfsdgasdfhjlkhajkfhlasjkhflksjhfsafsadfasdfadsfasdfasdfasdfasdfadsf"
        "asdfaasdfasdfdasdfasdfffasdfa");
    lv_obj_set_width(test, 200);
    lv_obj_set_style_max_height(test, 340, LV_PART_MAIN);
    lv_obj_update_layout(test);
    lv_label_set_long_mode(test, LV_LABEL_LONG_DOT);
```
I use lv_obj_update_layout to update the coordinates of the test obj, so in the lv_label_set_long_mode we can get the correct text.

**Comment by IAMMX** [comment]
This issue has been fixed by #5644, I think it can be closed.
